Rubocop rebase

barmintor opened the pull request:
@fcd1 @elo2112 I think this is going to need to be a group effort.

elo2112 commented:
Encountered this error when I ran the app and tried to load the default /digital_objects page:

NameError (undefined local variable or method `facet_limit' for DigitalObject::IndexAndSearch::FacetData:Module):
  app/models/concerns/digital_object/index_and_search/facet_data.rb:24:in `block (2 levels) in to_array'
  app/models/concerns/digital_object/index_and_search/facet_data.rb:23:in `each'
  app/models/concerns/digital_object/index_and_search/facet_data.rb:23:in `each_slice'
  app/models/concerns/digital_object/index_and_search/facet_data.rb:23:in `block in to_array'
  app/models/concerns/digital_object/index_and_search/facet_data.rb:15:in `each'
  app/models/concerns/digital_object/index_and_search/facet_data.rb:15:in `to_array'
  app/models/concerns/digital_object/index_and_search.rb:64:in `search'
  app/controllers/digital_objects_controller.rb:179:in `block (2 levels) in search'
  actionpack (4.2.4) lib/action_controller/metal/mime_responds.rb:217:in `call'
  actionpack (4.2.4) lib/action_controller/metal/mime_responds.rb:217:in `respond_to'
  app/controllers/digital_objects_controller.rb:177:in `search'
  actionpack (4.2.4) lib/action_controller/metal/implicit_render.rb:4:in `send_action'
  actionpack (4.2.4) lib/abstract_controller/base.rb:198:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/rendering.rb:10:in `process_action'
  actionpack (4.2.4) lib/abstract_controller/callbacks.rb:20:in `block in process_action'
  activesupport (4.2.4) lib/active_support/callbacks.rb:117:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:117:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:555:in `block (2 levels) in compile'
  activesupport (4.2.4) lib/active_support/callbacks.rb:505:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:505:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:92:in `__run_callbacks__'
  activesupport (4.2.4) lib/active_support/callbacks.rb:778:in `_run_process_action_callbacks'
  activesupport (4.2.4) lib/active_support/callbacks.rb:81:in `run_callbacks'
  actionpack (4.2.4) lib/abstract_controller/callbacks.rb:19:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/rescue.rb:29:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/instrumentation.rb:32:in `block in process_action'
  activesupport (4.2.4) lib/active_support/notifications.rb:164:in `block in instrument'
  activesupport (4.2.4) lib/active_support/notifications/instrumenter.rb:20:in `instrument'
  activesupport (4.2.4) lib/active_support/notifications.rb:164:in `instrument'
  actionpack (4.2.4) lib/action_controller/metal/instrumentation.rb:30:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/params_wrapper.rb:250:in `process_action'
  activerecord (4.2.4) lib/active_record/railties/controller_runtime.rb:18:in `process_action'
  actionpack (4.2.4) lib/abstract_controller/base.rb:137:in `process'
  actionview (4.2.4) lib/action_view/rendering.rb:30:in `process'
  actionpack (4.2.4) lib/action_controller/metal.rb:196:in `dispatch'
  actionpack (4.2.4) lib/action_controller/metal/rack_delegation.rb:13:in `dispatch'
  actionpack (4.2.4) lib/action_controller/metal.rb:237:in `block in action'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:76:in `call'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:76:in `dispatch'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:45:in `serve'
  actionpack (4.2.4) lib/action_dispatch/journey/router.rb:43:in `block in serve'
  actionpack (4.2.4) lib/action_dispatch/journey/router.rb:30:in `each'
  actionpack (4.2.4) lib/action_dispatch/journey/router.rb:30:in `serve'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:821:in `call'
  warden (1.2.3) lib/warden/manager.rb:35:in `block in call'
  warden (1.2.3) lib/warden/manager.rb:34:in `catch'
  warden (1.2.3) lib/warden/manager.rb:34:in `call'
  rack (1.6.4) lib/rack/etag.rb:24:in `call'
  rack (1.6.4) lib/rack/conditionalget.rb:38:in `call'
  rack (1.6.4) lib/rack/head.rb:13:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/params_parser.rb:27:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/flash.rb:260:in `call'
  rack (1.6.4) lib/rack/session/abstract/id.rb:225:in `context'
  rack (1.6.4) lib/rack/session/abstract/id.rb:220:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/cookies.rb:560:in `call'
  activerecord (4.2.4) lib/active_record/query_cache.rb:36:in `call'
  activerecord (4.2.4) lib/active_record/connection_adapters/abstract/connection_pool.rb:653:in `call'
  activerecord (4.2.4) lib/active_record/migration.rb:377:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/callbacks.rb:29:in `block in call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:88:in `__run_callbacks__'
  activesupport (4.2.4) lib/active_support/callbacks.rb:778:in `_run_call_callbacks'
  activesupport (4.2.4) lib/active_support/callbacks.rb:81:in `run_callbacks'
  actionpack (4.2.4) lib/action_dispatch/middleware/callbacks.rb:27:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/reloader.rb:73:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/remote_ip.rb:78:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
  web-console (2.2.1) lib/web_console/middleware.rb:39:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
  railties (4.2.4) lib/rails/rack/logger.rb:38:in `call_app'
  railties (4.2.4) lib/rails/rack/logger.rb:20:in `block in call'
  activesupport (4.2.4) lib/active_support/tagged_logging.rb:68:in `block in tagged'
  activesupport (4.2.4) lib/active_support/tagged_logging.rb:26:in `tagged'
  activesupport (4.2.4) lib/active_support/tagged_logging.rb:68:in `tagged'
  railties (4.2.4) lib/rails/rack/logger.rb:20:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/request_id.rb:21:in `call'
  rack (1.6.4) lib/rack/methodoverride.rb:22:in `call'
  rack (1.6.4) lib/rack/runtime.rb:18:in `call'
  activesupport (4.2.4) lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
  rack (1.6.4) lib/rack/lock.rb:17:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/static.rb:116:in `call'
  rack (1.6.4) lib/rack/sendfile.rb:113:in `call'
  railties (4.2.4) lib/rails/engine.rb:518:in `call'
  railties (4.2.4) lib/rails/application.rb:165:in `call'
  rack (1.6.4) lib/rack/content_length.rb:15:in `call'
  puma (2.11.1) lib/puma/server.rb:507:in `handle_request'
  puma (2.11.1) lib/puma/server.rb:375:in `process_client'
  puma (2.11.1) lib/puma/server.rb:262:in `block in run'
  puma (2.11.1) lib/puma/thread_pool.rb:104:in `call'
  puma (2.11.1) lib/puma/thread_pool.rb:104:in `block in spawn_thread'

elo2112 commented:
Also got this when trying to update a record via digital_objects_controller#update:

NoMethodError (undefined method `has_project_permission?' for #<User:0x007fcf7cc91280>):
  activemodel (4.2.4) lib/active_model/attribute_methods.rb:433:in `method_missing'
  app/controllers/application_controller.rb:104:in `block in check_project_permissions_and'
  app/controllers/application_controller.rb:103:in `each'
  app/controllers/application_controller.rb:103:in `check_project_permissions_and'
  app/controllers/application_controller.rb:86:in `require_project_permission!'
  app/controllers/digital_objects_controller.rb:312:in `require_appropriate_project_permissions!'
  activesupport (4.2.4) lib/active_support/callbacks.rb:432:in `block in make_lambda'
  activesupport (4.2.4) lib/active_support/callbacks.rb:164:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:164:in `block in halting'
  activesupport (4.2.4) lib/active_support/callbacks.rb:504:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:504:in `block in call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:504:in `each'
  activesupport (4.2.4) lib/active_support/callbacks.rb:504:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:92:in `__run_callbacks__'
  activesupport (4.2.4) lib/active_support/callbacks.rb:778:in `_run_process_action_callbacks'
  activesupport (4.2.4) lib/active_support/callbacks.rb:81:in `run_callbacks'
  actionpack (4.2.4) lib/abstract_controller/callbacks.rb:19:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/rescue.rb:29:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/instrumentation.rb:32:in `block in process_action'
  activesupport (4.2.4) lib/active_support/notifications.rb:164:in `block in instrument'
  activesupport (4.2.4) lib/active_support/notifications/instrumenter.rb:20:in `instrument'
  activesupport (4.2.4) lib/active_support/notifications.rb:164:in `instrument'
  actionpack (4.2.4) lib/action_controller/metal/instrumentation.rb:30:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/params_wrapper.rb:250:in `process_action'
  activerecord (4.2.4) lib/active_record/railties/controller_runtime.rb:18:in `process_action'
  actionpack (4.2.4) lib/abstract_controller/base.rb:137:in `process'
  actionview (4.2.4) lib/action_view/rendering.rb:30:in `process'
  actionpack (4.2.4) lib/action_controller/metal.rb:196:in `dispatch'
  actionpack (4.2.4) lib/action_controller/metal/rack_delegation.rb:13:in `dispatch'
  actionpack (4.2.4) lib/action_controller/metal.rb:237:in `block in action'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:76:in `call'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:76:in `dispatch'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:45:in `serve'
  actionpack (4.2.4) lib/action_dispatch/journey/router.rb:43:in `block in serve'
  actionpack (4.2.4) lib/action_dispatch/journey/router.rb:30:in `each'
  actionpack (4.2.4) lib/action_dispatch/journey/router.rb:30:in `serve'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:821:in `call'
  warden (1.2.3) lib/warden/manager.rb:35:in `block in call'
  warden (1.2.3) lib/warden/manager.rb:34:in `catch'
  warden (1.2.3) lib/warden/manager.rb:34:in `call'
  rack (1.6.4) lib/rack/etag.rb:24:in `call'
  rack (1.6.4) lib/rack/conditionalget.rb:38:in `call'
  rack (1.6.4) lib/rack/head.rb:13:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/params_parser.rb:27:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/flash.rb:260:in `call'
  rack (1.6.4) lib/rack/session/abstract/id.rb:225:in `context'
  rack (1.6.4) lib/rack/session/abstract/id.rb:220:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/cookies.rb:560:in `call'
  activerecord (4.2.4) lib/active_record/query_cache.rb:36:in `call'
  activerecord (4.2.4) lib/active_record/connection_adapters/abstract/connection_pool.rb:653:in `call'
  activerecord (4.2.4) lib/active_record/migration.rb:377:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/callbacks.rb:29:in `block in call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:88:in `__run_callbacks__'
  activesupport (4.2.4) lib/active_support/callbacks.rb:778:in `_run_call_callbacks'
  activesupport (4.2.4) lib/active_support/callbacks.rb:81:in `run_callbacks'
  actionpack (4.2.4) lib/action_dispatch/middleware/callbacks.rb:27:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/reloader.rb:73:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/remote_ip.rb:78:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
  web-console (2.2.1) lib/web_console/middleware.rb:39:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
  railties (4.2.4) lib/rails/rack/logger.rb:38:in `call_app'
  railties (4.2.4) lib/rails/rack/logger.rb:20:in `block in call'
  activesupport (4.2.4) lib/active_support/tagged_logging.rb:68:in `block in tagged'
  activesupport (4.2.4) lib/active_support/tagged_logging.rb:26:in `tagged'
  activesupport (4.2.4) lib/active_support/tagged_logging.rb:68:in `tagged'
  railties (4.2.4) lib/rails/rack/logger.rb:20:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/request_id.rb:21:in `call'
  rack (1.6.4) lib/rack/methodoverride.rb:22:in `call'
  rack (1.6.4) lib/rack/runtime.rb:18:in `call'
  activesupport (4.2.4) lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
  rack (1.6.4) lib/rack/lock.rb:17:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/static.rb:116:in `call'
  rack (1.6.4) lib/rack/sendfile.rb:113:in `call'
  railties (4.2.4) lib/rails/engine.rb:518:in `call'
  railties (4.2.4) lib/rails/application.rb:165:in `call'
  rack (1.6.4) lib/rack/content_length.rb:15:in `call'
  puma (2.11.1) lib/puma/server.rb:507:in `handle_request'
  puma (2.11.1) lib/puma/server.rb:375:in `process_client'
  puma (2.11.1) lib/puma/server.rb:262:in `block in run'
  puma (2.11.1) lib/puma/thread_pool.rb:104:in `call'
  puma (2.11.1) lib/puma/thread_pool.rb:104:in `block in spawn_thread'

elo2112 commented:
Mentioning @fcd1 so that he becomes a watcher on this pull request.

elo2112 commented:
Never mind about my @fcd1 mention.  I see that @barmintor already did that at the top of this request.

fcd1 commented:
The changes shown in branch fcd1-rubocop-rebase-debug seem to help with the facet-related issues. Regarding update: I'm able to edit and save an object without any issues via the GUI, logged in as hyacinth-admin@library.columbia.edu.

elo2112 commented:
I'm going to close this pull request in favor of a future one from @fcd1 that includes his latest changes from the above comment.

barmintor commented:
no

On Wed, May 18, 2016 at 6:51 PM, Eric O <notifications@github.com> wrote:

> I'm going to close this pull request in favor of a future one from @fcd1
> <https://github.com/fcd1> that includes his latest changes from the above
> comment.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/cul/hyacinth/pull/20#issuecomment-220181520>
>

barmintor commented:
@fcd1 ok, I see part of what you mean. I don't think it fixes the other thing @elo2112 reported, so let's pull that change into this PR (and avoid rebranching and re-PRing more than necessary).

barmintor commented:
ok, Fred's changes are pulled in. If we're adding commits to the end of a
branch, you can always send a PR to the rubocop branch! It'll help minimize
the amount of branch chasing we have to do. Thanks @fcd1!

On Wed, May 18, 2016 at 7:18 PM, Benjamin Armintor <armintor@gmail.com>
wrote:

> no
>
> On Wed, May 18, 2016 at 6:51 PM, Eric O <notifications@github.com> wrote:
>
>> I'm going to close this pull request in favor of a future one from @fcd1
>> <https://github.com/fcd1> that includes his latest changes from the
>> above comment.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/cul/hyacinth/pull/20#issuecomment-220181520>
>>
>
>

elo2112 commented:
Much better solution than a new pull request.  Thanks, @barmintor.

fcd1 commented:
I concur with all the above.

elo2112 commented:
Just tried running the app again.  I haven't encountered any other issues so far as an admin user, but I'm running into these two problems when logged in as a NON-admin user:

1) I can view search results, but can't view an item-level page (digital_objects_controller#data_for_editor):

``NoMethodError (undefined method `has_project_permission?' for #<User:0x007fce70a5f320>):
  activemodel (4.2.4) lib/active_model/attribute_methods.rb:433:in `method_missing'
  app/controllers/application_controller.rb:104:in `block in check_project_permissions_and'
  app/controllers/application_controller.rb:103:in `each'
  app/controllers/application_controller.rb:103:in `check_project_permissions_and'
  app/controllers/application_controller.rb:86:in `require_project_permission!'
  app/controllers/digital_objects_controller.rb:304:in `require_appropriate_project_permissions!'
  activesupport (4.2.4) lib/active_support/callbacks.rb:432:in `block in make_lambda'
  activesupport (4.2.4) lib/active_support/callbacks.rb:164:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:164:in `block in halting'
  activesupport (4.2.4) lib/active_support/callbacks.rb:504:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:504:in `block in call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:504:in `each'
  activesupport (4.2.4) lib/active_support/callbacks.rb:504:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:92:in `__run_callbacks__'
  activesupport (4.2.4) lib/active_support/callbacks.rb:778:in `_run_process_action_callbacks'
  activesupport (4.2.4) lib/active_support/callbacks.rb:81:in `run_callbacks'
  actionpack (4.2.4) lib/abstract_controller/callbacks.rb:19:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/rescue.rb:29:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/instrumentation.rb:32:in `block in process_action'
  activesupport (4.2.4) lib/active_support/notifications.rb:164:in `block in instrument'
  activesupport (4.2.4) lib/active_support/notifications/instrumenter.rb:20:in `instrument'
  activesupport (4.2.4) lib/active_support/notifications.rb:164:in `instrument'
  actionpack (4.2.4) lib/action_controller/metal/instrumentation.rb:30:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/params_wrapper.rb:250:in `process_action'
  activerecord (4.2.4) lib/active_record/railties/controller_runtime.rb:18:in `process_action'
  actionpack (4.2.4) lib/abstract_controller/base.rb:137:in `process'
  actionview (4.2.4) lib/action_view/rendering.rb:30:in `process'
  actionpack (4.2.4) lib/action_controller/metal.rb:196:in `dispatch'
  actionpack (4.2.4) lib/action_controller/metal/rack_delegation.rb:13:in `dispatch'
  actionpack (4.2.4) lib/action_controller/metal.rb:237:in `block in action'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:76:in `call'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:76:in `dispatch'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:45:in `serve'
  actionpack (4.2.4) lib/action_dispatch/journey/router.rb:43:in `block in serve'
  actionpack (4.2.4) lib/action_dispatch/journey/router.rb:30:in `each'
  actionpack (4.2.4) lib/action_dispatch/journey/router.rb:30:in `serve'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:821:in `call'
  warden (1.2.3) lib/warden/manager.rb:35:in `block in call'
  warden (1.2.3) lib/warden/manager.rb:34:in `catch'
  warden (1.2.3) lib/warden/manager.rb:34:in `call'
  rack (1.6.4) lib/rack/etag.rb:24:in `call'
  rack (1.6.4) lib/rack/conditionalget.rb:38:in `call'
  rack (1.6.4) lib/rack/head.rb:13:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/params_parser.rb:27:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/flash.rb:260:in `call'
  rack (1.6.4) lib/rack/session/abstract/id.rb:225:in `context'
  rack (1.6.4) lib/rack/session/abstract/id.rb:220:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/cookies.rb:560:in `call'
  activerecord (4.2.4) lib/active_record/query_cache.rb:36:in `call'
  activerecord (4.2.4) lib/active_record/connection_adapters/abstract/connection_pool.rb:653:in `call'
  activerecord (4.2.4) lib/active_record/migration.rb:377:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/callbacks.rb:29:in `block in call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:88:in `__run_callbacks__'
  activesupport (4.2.4) lib/active_support/callbacks.rb:778:in `_run_call_callbacks'
  activesupport (4.2.4) lib/active_support/callbacks.rb:81:in `run_callbacks'
  actionpack (4.2.4) lib/action_dispatch/middleware/callbacks.rb:27:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/reloader.rb:73:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/remote_ip.rb:78:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
  web-console (2.2.1) lib/web_console/middleware.rb:39:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
  railties (4.2.4) lib/rails/rack/logger.rb:38:in `call_app'
  railties (4.2.4) lib/rails/rack/logger.rb:20:in `block in call'
  activesupport (4.2.4) lib/active_support/tagged_logging.rb:68:in `block in tagged'
  activesupport (4.2.4) lib/active_support/tagged_logging.rb:26:in `tagged'
  activesupport (4.2.4) lib/active_support/tagged_logging.rb:68:in `tagged'
  railties (4.2.4) lib/rails/rack/logger.rb:20:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/request_id.rb:21:in `call'
  rack (1.6.4) lib/rack/methodoverride.rb:22:in `call'
  rack (1.6.4) lib/rack/runtime.rb:18:in `call'
  activesupport (4.2.4) lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
  rack (1.6.4) lib/rack/lock.rb:17:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/static.rb:116:in `call'
  rack (1.6.4) lib/rack/sendfile.rb:113:in `call'
  railties (4.2.4) lib/rails/engine.rb:518:in `call'
  railties (4.2.4) lib/rails/application.rb:165:in `call'
  rack (1.6.4) lib/rack/content_length.rb:15:in `call'
  puma (2.11.1) lib/puma/server.rb:507:in `handle_request'
  puma (2.11.1) lib/puma/server.rb:375:in `process_client'
  puma (2.11.1) lib/puma/server.rb:262:in `block in run'
  puma (2.11.1) lib/puma/thread_pool.rb:104:in `call'
  puma (2.11.1) lib/puma/thread_pool.rb:104:in `block in spawn_thread'``

2. And I still can't update a digital object, but it's a different error now:

``NoMethodError (undefined method `fetch' for "false":String):
  app/models/concerns/digital_object/index_and_search/facet_data.rb:12:in `to_array'
  app/models/concerns/digital_object/index_and_search.rb:64:in `search'
  app/controllers/digital_objects_controller.rb:179:in `block (2 levels) in search'
  actionpack (4.2.4) lib/action_controller/metal/mime_responds.rb:217:in `call'
  actionpack (4.2.4) lib/action_controller/metal/mime_responds.rb:217:in `respond_to'
  app/controllers/digital_objects_controller.rb:177:in `search'
  actionpack (4.2.4) lib/action_controller/metal/implicit_render.rb:4:in `send_action'
  actionpack (4.2.4) lib/abstract_controller/base.rb:198:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/rendering.rb:10:in `process_action'
  actionpack (4.2.4) lib/abstract_controller/callbacks.rb:20:in `block in process_action'
  activesupport (4.2.4) lib/active_support/callbacks.rb:117:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:117:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:555:in `block (2 levels) in compile'
  activesupport (4.2.4) lib/active_support/callbacks.rb:505:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:505:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:92:in `__run_callbacks__'
  activesupport (4.2.4) lib/active_support/callbacks.rb:778:in `_run_process_action_callbacks'
  activesupport (4.2.4) lib/active_support/callbacks.rb:81:in `run_callbacks'
  actionpack (4.2.4) lib/abstract_controller/callbacks.rb:19:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/rescue.rb:29:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/instrumentation.rb:32:in `block in process_action'
  activesupport (4.2.4) lib/active_support/notifications.rb:164:in `block in instrument'
  activesupport (4.2.4) lib/active_support/notifications/instrumenter.rb:20:in `instrument'
  activesupport (4.2.4) lib/active_support/notifications.rb:164:in `instrument'
  actionpack (4.2.4) lib/action_controller/metal/instrumentation.rb:30:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/params_wrapper.rb:250:in `process_action'
  activerecord (4.2.4) lib/active_record/railties/controller_runtime.rb:18:in `process_action'
  actionpack (4.2.4) lib/abstract_controller/base.rb:137:in `process'
  actionview (4.2.4) lib/action_view/rendering.rb:30:in `process'
  actionpack (4.2.4) lib/action_controller/metal.rb:196:in `dispatch'
  actionpack (4.2.4) lib/action_controller/metal/rack_delegation.rb:13:in `dispatch'
  actionpack (4.2.4) lib/action_controller/metal.rb:237:in `block in action'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:76:in `call'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:76:in `dispatch'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:45:in `serve'
  actionpack (4.2.4) lib/action_dispatch/journey/router.rb:43:in `block in serve'
  actionpack (4.2.4) lib/action_dispatch/journey/router.rb:30:in `each'
  actionpack (4.2.4) lib/action_dispatch/journey/router.rb:30:in `serve'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:821:in `call'
  warden (1.2.3) lib/warden/manager.rb:35:in `block in call'
  warden (1.2.3) lib/warden/manager.rb:34:in `catch'
  warden (1.2.3) lib/warden/manager.rb:34:in `call'
  rack (1.6.4) lib/rack/etag.rb:24:in `call'
  rack (1.6.4) lib/rack/conditionalget.rb:38:in `call'
  rack (1.6.4) lib/rack/head.rb:13:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/params_parser.rb:27:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/flash.rb:260:in `call'
  rack (1.6.4) lib/rack/session/abstract/id.rb:225:in `context'
  rack (1.6.4) lib/rack/session/abstract/id.rb:220:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/cookies.rb:560:in `call'
  activerecord (4.2.4) lib/active_record/query_cache.rb:36:in `call'
  activerecord (4.2.4) lib/active_record/connection_adapters/abstract/connection_pool.rb:653:in `call'
  activerecord (4.2.4) lib/active_record/migration.rb:377:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/callbacks.rb:29:in `block in call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:88:in `__run_callbacks__'
  activesupport (4.2.4) lib/active_support/callbacks.rb:778:in `_run_call_callbacks'
  activesupport (4.2.4) lib/active_support/callbacks.rb:81:in `run_callbacks'
  actionpack (4.2.4) lib/action_dispatch/middleware/callbacks.rb:27:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/reloader.rb:73:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/remote_ip.rb:78:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
  web-console (2.2.1) lib/web_console/middleware.rb:39:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
  railties (4.2.4) lib/rails/rack/logger.rb:38:in `call_app'
  railties (4.2.4) lib/rails/rack/logger.rb:20:in `block in call'
  activesupport (4.2.4) lib/active_support/tagged_logging.rb:68:in `block in tagged'
  activesupport (4.2.4) lib/active_support/tagged_logging.rb:26:in `tagged'
  activesupport (4.2.4) lib/active_support/tagged_logging.rb:68:in `tagged'
  railties (4.2.4) lib/rails/rack/logger.rb:20:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/request_id.rb:21:in `call'
  rack (1.6.4) lib/rack/methodoverride.rb:22:in `call'
  rack (1.6.4) lib/rack/runtime.rb:18:in `call'
  activesupport (4.2.4) lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
  rack (1.6.4) lib/rack/lock.rb:17:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/static.rb:116:in `call'
  rack (1.6.4) lib/rack/sendfile.rb:113:in `call'
  railties (4.2.4) lib/rails/engine.rb:518:in `call'
  railties (4.2.4) lib/rails/application.rb:165:in `call'
  rack (1.6.4) lib/rack/content_length.rb:15:in `call'
  puma (2.11.1) lib/puma/server.rb:507:in `handle_request'
  puma (2.11.1) lib/puma/server.rb:375:in `process_client'
  puma (2.11.1) lib/puma/server.rb:262:in `block in run'
  puma (2.11.1) lib/puma/thread_pool.rb:104:in `call'
  puma (2.11.1) lib/puma/thread_pool.rb:104:in `block in spawn_thread'``

barmintor commented:
It's possible that I moved a permissions method to the User model that got
dropped in the rebase- I'll check tomorrow at the office, I still have the
original branch there. The fetch/false thing might come from a
rearrangement of parameters in one those new commits.

On Wed, May 18, 2016 at 7:53 PM, Eric O <notifications@github.com> wrote:

> Just tried running the app again. I haven't encountered any other issues
> so far as an admin user, but I'm running into these two problems when
> logged in as a NON-admin user:
>
> 1) I can view search results, but can't view an item-level page
> (digital_objects_controller#data_for_editor):
>
> NoMethodError (undefined method `has_project_permission?' for
> #<User:0x007fce70a5f320>):
> activemodel (4.2.4) lib/active_model/attribute_methods.rb:433:in
> `method_missing'
> app/controllers/application_controller.rb:104:in `block in
> check_project_permissions_and'
> app/controllers/application_controller.rb:103:in `each'
> app/controllers/application_controller.rb:103:in
> `check_project_permissions_and'
> app/controllers/application_controller.rb:86:in
> `require_project_permission!'
> app/controllers/digital_objects_controller.rb:304:in
> `require_appropriate_project_permissions!'
> activesupport (4.2.4) lib/active_support/callbacks.rb:432:in `block in
> make_lambda'
> activesupport (4.2.4) lib/active_support/callbacks.rb:164:in `call'
> activesupport (4.2.4) lib/active_support/callbacks.rb:164:in `block in
> halting'
> activesupport (4.2.4) lib/active_support/callbacks.rb:504:in `call'
> activesupport (4.2.4) lib/active_support/callbacks.rb:504:in `block in
> call'
> activesupport (4.2.4) lib/active_support/callbacks.rb:504:in `each'
> activesupport (4.2.4) lib/active_support/callbacks.rb:504:in `call'
> activesupport (4.2.4) lib/active_support/callbacks.rb:92:in
> `__run_callbacks__'
> activesupport (4.2.4) lib/active_support/callbacks.rb:778:in
> `_run_process_action_callbacks'
> activesupport (4.2.4) lib/active_support/callbacks.rb:81:in `run_callbacks'
> actionpack (4.2.4) lib/abstract_controller/callbacks.rb:19:in
> `process_action'
> actionpack (4.2.4) lib/action_controller/metal/rescue.rb:29:in
> `process_action'
> actionpack (4.2.4) lib/action_controller/metal/instrumentation.rb:32:in
> `block in process_action'
> activesupport (4.2.4) lib/active_support/notifications.rb:164:in `block in
> instrument'
> activesupport (4.2.4)
> lib/active_support/notifications/instrumenter.rb:20:in `instrument'
> activesupport (4.2.4) lib/active_support/notifications.rb:164:in
> `instrument'
> actionpack (4.2.4) lib/action_controller/metal/instrumentation.rb:30:in
> `process_action'
> actionpack (4.2.4) lib/action_controller/metal/params_wrapper.rb:250:in
> `process_action'
> activerecord (4.2.4)
> lib/active_record/railties/controller_runtime.rb:18:in `process_action'
> actionpack (4.2.4) lib/abstract_controller/base.rb:137:in `process'
> actionview (4.2.4) lib/action_view/rendering.rb:30:in `process'
> actionpack (4.2.4) lib/action_controller/metal.rb:196:in `dispatch'
> actionpack (4.2.4) lib/action_controller/metal/rack_delegation.rb:13:in
> `dispatch'
> actionpack (4.2.4) lib/action_controller/metal.rb:237:in `block in action'
> actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:76:in `call'
> actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:76:in
> `dispatch'
> actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:45:in `serve'
> actionpack (4.2.4) lib/action_dispatch/journey/router.rb:43:in `block in
> serve'
> actionpack (4.2.4) lib/action_dispatch/journey/router.rb:30:in `each'
> actionpack (4.2.4) lib/action_dispatch/journey/router.rb:30:in `serve'
> actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:821:in `call'
> warden (1.2.3) lib/warden/manager.rb:35:in `block in call'
> warden (1.2.3) lib/warden/manager.rb:34:in `catch'
> warden (1.2.3) lib/warden/manager.rb:34:in `call'
> rack (1.6.4) lib/rack/etag.rb:24:in `call'
> rack (1.6.4) lib/rack/conditionalget.rb:38:in `call'
> rack (1.6.4) lib/rack/head.rb:13:in `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/params_parser.rb:27:in
> `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/flash.rb:260:in `call'
> rack (1.6.4) lib/rack/session/abstract/id.rb:225:in `context'
> rack (1.6.4) lib/rack/session/abstract/id.rb:220:in `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/cookies.rb:560:in `call'
> activerecord (4.2.4) lib/active_record/query_cache.rb:36:in `call'
> activerecord (4.2.4)
> lib/active_record/connection_adapters/abstract/connection_pool.rb:653:in
> `call'
> activerecord (4.2.4) lib/active_record/migration.rb:377:in `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/callbacks.rb:29:in
> `block in call'
> activesupport (4.2.4) lib/active_support/callbacks.rb:88:in
> `__run_callbacks__'
> activesupport (4.2.4) lib/active_support/callbacks.rb:778:in
> `_run_call_callbacks'
> activesupport (4.2.4) lib/active_support/callbacks.rb:81:in `run_callbacks'
> actionpack (4.2.4) lib/action_dispatch/middleware/callbacks.rb:27:in `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/reloader.rb:73:in `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/remote_ip.rb:78:in `call'
> actionpack (4.2.4)
> lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
> web-console (2.2.1) lib/web_console/middleware.rb:39:in `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/show_exceptions.rb:30:in
> `call'
> railties (4.2.4) lib/rails/rack/logger.rb:38:in `call_app'
> railties (4.2.4) lib/rails/rack/logger.rb:20:in `block in call'
> activesupport (4.2.4) lib/active_support/tagged_logging.rb:68:in `block in
> tagged'
> activesupport (4.2.4) lib/active_support/tagged_logging.rb:26:in `tagged'
> activesupport (4.2.4) lib/active_support/tagged_logging.rb:68:in `tagged'
> railties (4.2.4) lib/rails/rack/logger.rb:20:in `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/request_id.rb:21:in
> `call'
> rack (1.6.4) lib/rack/methodoverride.rb:22:in `call'
> rack (1.6.4) lib/rack/runtime.rb:18:in `call'
> activesupport (4.2.4)
> lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
> rack (1.6.4) lib/rack/lock.rb:17:in `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/static.rb:116:in `call'
> rack (1.6.4) lib/rack/sendfile.rb:113:in `call'
> railties (4.2.4) lib/rails/engine.rb:518:in `call'
> railties (4.2.4) lib/rails/application.rb:165:in `call'
> rack (1.6.4) lib/rack/content_length.rb:15:in `call'
> puma (2.11.1) lib/puma/server.rb:507:in `handle_request'
> puma (2.11.1) lib/puma/server.rb:375:in `process_client'
> puma (2.11.1) lib/puma/server.rb:262:in `block in run'
> puma (2.11.1) lib/puma/thread_pool.rb:104:in `call'
> puma (2.11.1) lib/puma/thread_pool.rb:104:in `block in spawn_thread'
>
>    1. And I still can't update a digital object, but it's a different
>    error now:
>
> NoMethodError (undefined method `fetch' for "false":String):
> app/models/concerns/digital_object/index_and_search/facet_data.rb:12:in
> `to_array'
> app/models/concerns/digital_object/index_and_search.rb:64:in `search'
> app/controllers/digital_objects_controller.rb:179:in `block (2 levels) in
> search'
> actionpack (4.2.4) lib/action_controller/metal/mime_responds.rb:217:in
> `call'
> actionpack (4.2.4) lib/action_controller/metal/mime_responds.rb:217:in
> `respond_to'
> app/controllers/digital_objects_controller.rb:177:in `search'
> actionpack (4.2.4) lib/action_controller/metal/implicit_render.rb:4:in
> `send_action'
> actionpack (4.2.4) lib/abstract_controller/base.rb:198:in `process_action'
> actionpack (4.2.4) lib/action_controller/metal/rendering.rb:10:in
> `process_action'
> actionpack (4.2.4) lib/abstract_controller/callbacks.rb:20:in `block in
> process_action'
> activesupport (4.2.4) lib/active_support/callbacks.rb:117:in `call'
> activesupport (4.2.4) lib/active_support/callbacks.rb:117:in `call'
> activesupport (4.2.4) lib/active_support/callbacks.rb:555:in `block (2
> levels) in compile'
> activesupport (4.2.4) lib/active_support/callbacks.rb:505:in `call'
> activesupport (4.2.4) lib/active_support/callbacks.rb:505:in `call'
> activesupport (4.2.4) lib/active_support/callbacks.rb:92:in
> `__run_callbacks__'
> activesupport (4.2.4) lib/active_support/callbacks.rb:778:in
> `_run_process_action_callbacks'
> activesupport (4.2.4) lib/active_support/callbacks.rb:81:in `run_callbacks'
> actionpack (4.2.4) lib/abstract_controller/callbacks.rb:19:in
> `process_action'
> actionpack (4.2.4) lib/action_controller/metal/rescue.rb:29:in
> `process_action'
> actionpack (4.2.4) lib/action_controller/metal/instrumentation.rb:32:in
> `block in process_action'
> activesupport (4.2.4) lib/active_support/notifications.rb:164:in `block in
> instrument'
> activesupport (4.2.4)
> lib/active_support/notifications/instrumenter.rb:20:in `instrument'
> activesupport (4.2.4) lib/active_support/notifications.rb:164:in
> `instrument'
> actionpack (4.2.4) lib/action_controller/metal/instrumentation.rb:30:in
> `process_action'
> actionpack (4.2.4) lib/action_controller/metal/params_wrapper.rb:250:in
> `process_action'
> activerecord (4.2.4)
> lib/active_record/railties/controller_runtime.rb:18:in `process_action'
> actionpack (4.2.4) lib/abstract_controller/base.rb:137:in `process'
> actionview (4.2.4) lib/action_view/rendering.rb:30:in `process'
> actionpack (4.2.4) lib/action_controller/metal.rb:196:in `dispatch'
> actionpack (4.2.4) lib/action_controller/metal/rack_delegation.rb:13:in
> `dispatch'
> actionpack (4.2.4) lib/action_controller/metal.rb:237:in `block in action'
> actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:76:in `call'
> actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:76:in
> `dispatch'
> actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:45:in `serve'
> actionpack (4.2.4) lib/action_dispatch/journey/router.rb:43:in `block in
> serve'
> actionpack (4.2.4) lib/action_dispatch/journey/router.rb:30:in `each'
> actionpack (4.2.4) lib/action_dispatch/journey/router.rb:30:in `serve'
> actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:821:in `call'
> warden (1.2.3) lib/warden/manager.rb:35:in `block in call'
> warden (1.2.3) lib/warden/manager.rb:34:in `catch'
> warden (1.2.3) lib/warden/manager.rb:34:in `call'
> rack (1.6.4) lib/rack/etag.rb:24:in `call'
> rack (1.6.4) lib/rack/conditionalget.rb:38:in `call'
> rack (1.6.4) lib/rack/head.rb:13:in `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/params_parser.rb:27:in
> `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/flash.rb:260:in `call'
> rack (1.6.4) lib/rack/session/abstract/id.rb:225:in `context'
> rack (1.6.4) lib/rack/session/abstract/id.rb:220:in `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/cookies.rb:560:in `call'
> activerecord (4.2.4) lib/active_record/query_cache.rb:36:in `call'
> activerecord (4.2.4)
> lib/active_record/connection_adapters/abstract/connection_pool.rb:653:in
> `call'
> activerecord (4.2.4) lib/active_record/migration.rb:377:in `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/callbacks.rb:29:in
> `block in call'
> activesupport (4.2.4) lib/active_support/callbacks.rb:88:in
> `__run_callbacks__'
> activesupport (4.2.4) lib/active_support/callbacks.rb:778:in
> `_run_call_callbacks'
> activesupport (4.2.4) lib/active_support/callbacks.rb:81:in `run_callbacks'
> actionpack (4.2.4) lib/action_dispatch/middleware/callbacks.rb:27:in `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/reloader.rb:73:in `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/remote_ip.rb:78:in `call'
> actionpack (4.2.4)
> lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
> web-console (2.2.1) lib/web_console/middleware.rb:39:in `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/show_exceptions.rb:30:in
> `call'
> railties (4.2.4) lib/rails/rack/logger.rb:38:in `call_app'
> railties (4.2.4) lib/rails/rack/logger.rb:20:in `block in call'
> activesupport (4.2.4) lib/active_support/tagged_logging.rb:68:in `block in
> tagged'
> activesupport (4.2.4) lib/active_support/tagged_logging.rb:26:in `tagged'
> activesupport (4.2.4) lib/active_support/tagged_logging.rb:68:in `tagged'
> railties (4.2.4) lib/rails/rack/logger.rb:20:in `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/request_id.rb:21:in
> `call'
> rack (1.6.4) lib/rack/methodoverride.rb:22:in `call'
> rack (1.6.4) lib/rack/runtime.rb:18:in `call'
> activesupport (4.2.4)
> lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
> rack (1.6.4) lib/rack/lock.rb:17:in `call'
> actionpack (4.2.4) lib/action_dispatch/middleware/static.rb:116:in `call'
> rack (1.6.4) lib/rack/sendfile.rb:113:in `call'
> railties (4.2.4) lib/rails/engine.rb:518:in `call'
> railties (4.2.4) lib/rails/application.rb:165:in `call'
> rack (1.6.4) lib/rack/content_length.rb:15:in `call'
> puma (2.11.1) lib/puma/server.rb:507:in `handle_request'
> puma (2.11.1) lib/puma/server.rb:375:in `process_client'
> puma (2.11.1) lib/puma/server.rb:262:in `block in run'
> puma (2.11.1) lib/puma/thread_pool.rb:104:in `call'
> puma (2.11.1) lib/puma/thread_pool.rb:104:in `block in spawn_thread'
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/cul/hyacinth/pull/20#issuecomment-220191617>
>

fcd1 commented:
Yes, I did rearrange the parameters in DigitalObject::IndexAndSearch#search in order to get rid of an argument (facet_params set to false) in a few calls. I'll look at the fetch issue tomorrow.

fcd1 commented:
fetch issue: facet_params is used in conflicting ways: before the refactor, facet_params was set to false to indicate that these params were not set, and to use defaults instead. In the refactored code, facet_params defaults to {}, and Hash#fetch is used to set the defaults. I prefer the latter, and my committed code changes reflect that choice (which I'm willing to change if there are concerns (not the rails type)). I looked for all the places where facet_params was set to false, and set it to {} instead. However, from the error #2 encountered by Eric, I obviously missed at least one spot -- I'm guessing this is being set indirectly, i.e. foo is set to false somewhere, then facet_params is set to foo somewhere else.

barmintor commented:
:+1:

On Thu, May 19, 2016 at 8:50 AM, fcd1 <notifications@github.com> wrote:

> fetch issue: facet_params is used in conflicting ways: before the
> refactor, facet_params was set to false to indicate that these params were
> not set, and to use defaults instead. In the refactored code, facet_params
> defaults to {}, and Hash#fetch is used to set the defaults. I prefer the
> latter, and my committed code changes reflect that choice (which I'm
> willing to change if there are concerns (not the rails type)). I looked for
> all the places where facet_params was set to false, and set it to {}
> instead. However, from the error #2
> <https://github.com/cul/hyacinth/issues/2> encountered by Eric, I
> obviously missed at least one spot -- I'm guessing this is being set
> indirectly, i.e. foo is set to false somewhere, then facet_params is set to
> foo somewhere else.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/cul/hyacinth/pull/20#issuecomment-220314431>
>

fcd1 commented:
Permissions issues: #has_project_permission? has been renamed to #permitted_in_project?. I'll replace the remaining calls (2) to the old method with calls to the new method.

elo2112 commented:
Great!  Thanks!

elo2112 commented:
Update:

The digital object update works in the Hyacinth GUI, but I'm getting the digital object update error I mentioned when I try to run an update via DurstVoyagerHyacinthSync java app (https://github.com/cul/durst_voyager_hyacinth_sync).  Reason is, I'm still sending {"facet" : false} in my JSON update.  We'll have to be careful about changes that change the REST API.  (And even better, we'll want to add tests for all REST API features so that we'll know when we break the API.)  I'm going to update the Durst app.

One other problem though -- while looking through the PR changes though, I noticed this in the diff:

``def set_facet_params_for(facet_params, solr_params, dynamic_field_string_keys_to_dynamic_fields)
 +      # facet_params are for specifying how you want to RECEIVE facets.
 +      # This has nothing to do with using the facet feature to apply facet filters.
 +      if facet_params.to_s == "false"
 +        solr_params['facet'] = false
 +      else``

We're still doing a comparison with "false", even though facet_params now defaults to an empty hash in the search method.

fcd1 commented:
Trying to generate MODS XML (via the MODS XML link on an Item page) will throw an error:

TypeError (no implicit conversion of String into Integer):
  app/models/concerns/digital_object/xml_datastream_rendering.rb:22:in `[]'
  app/models/concerns/digital_object/xml_datastream_rendering.rb:22:in `should_render?'
  app/models/concerns/digital_object/xml_datastream_rendering.rb:46:in `render_xml_translation_with_data'
  app/models/concerns/digital_object/xml_datastream_rendering.rb:75:in `block (3 levels) in render_xml_translation_with_data'
  app/models/concerns/digital_object/xml_datastream_rendering.rb:74:in `each'
  app/models/concerns/digital_object/xml_datastream_rendering.rb:74:in `block (2 levels) in render_xml_translation_with_data'
  app/models/concerns/digital_object/xml_datastream_rendering.rb:73:in `each'
  app/models/concerns/digital_object/xml_datastream_rendering.rb:73:in `block in render_xml_translation_with_data'
  app/models/concerns/digital_object/xml_datastream_rendering.rb:68:in `each'
  app/models/concerns/digital_object/xml_datastream_rendering.rb:68:in `render_xml_translation_with_data'
  app/models/concerns/digital_object/xml_datastream_rendering.rb:12:in `render_xml_datastream'
  app/controllers/digital_objects_controller.rb:206:in `mods'
  actionpack (4.2.4) lib/action_controller/metal/implicit_render.rb:4:in `send_action'
  actionpack (4.2.4) lib/abstract_controller/base.rb:198:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/rendering.rb:10:in `process_action'
  actionpack (4.2.4) lib/abstract_controller/callbacks.rb:20:in `block in process_action'
  activesupport (4.2.4) lib/active_support/callbacks.rb:117:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:117:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:555:in `block (2 levels) in compile'
  activesupport (4.2.4) lib/active_support/callbacks.rb:505:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:505:in `call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:92:in `__run_callbacks__'
  activesupport (4.2.4) lib/active_support/callbacks.rb:778:in `_run_process_action_callbacks'
  activesupport (4.2.4) lib/active_support/callbacks.rb:81:in `run_callbacks'
  actionpack (4.2.4) lib/abstract_controller/callbacks.rb:19:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/rescue.rb:29:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/instrumentation.rb:32:in `block in process_action'
  activesupport (4.2.4) lib/active_support/notifications.rb:164:in `block in instrument'
  activesupport (4.2.4) lib/active_support/notifications/instrumenter.rb:20:in `instrument'
  activesupport (4.2.4) lib/active_support/notifications.rb:164:in `instrument'
  actionpack (4.2.4) lib/action_controller/metal/instrumentation.rb:30:in `process_action'
  actionpack (4.2.4) lib/action_controller/metal/params_wrapper.rb:250:in `process_action'
  activerecord (4.2.4) lib/active_record/railties/controller_runtime.rb:18:in `process_action'
  actionpack (4.2.4) lib/abstract_controller/base.rb:137:in `process'
  actionview (4.2.4) lib/action_view/rendering.rb:30:in `process'
  actionpack (4.2.4) lib/action_controller/metal.rb:196:in `dispatch'
  actionpack (4.2.4) lib/action_controller/metal/rack_delegation.rb:13:in `dispatch'
  actionpack (4.2.4) lib/action_controller/metal.rb:237:in `block in action'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:76:in `call'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:76:in `dispatch'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:45:in `serve'
  actionpack (4.2.4) lib/action_dispatch/journey/router.rb:43:in `block in serve'
  actionpack (4.2.4) lib/action_dispatch/journey/router.rb:30:in `each'
  actionpack (4.2.4) lib/action_dispatch/journey/router.rb:30:in `serve'
  actionpack (4.2.4) lib/action_dispatch/routing/route_set.rb:821:in `call'
  warden (1.2.3) lib/warden/manager.rb:35:in `block in call'
  warden (1.2.3) lib/warden/manager.rb:34:in `catch'
  warden (1.2.3) lib/warden/manager.rb:34:in `call'
  rack (1.6.4) lib/rack/etag.rb:24:in `call'
  rack (1.6.4) lib/rack/conditionalget.rb:25:in `call'
  rack (1.6.4) lib/rack/head.rb:13:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/params_parser.rb:27:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/flash.rb:260:in `call'
  rack (1.6.4) lib/rack/session/abstract/id.rb:225:in `context'
  rack (1.6.4) lib/rack/session/abstract/id.rb:220:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/cookies.rb:560:in `call'
  activerecord (4.2.4) lib/active_record/query_cache.rb:36:in `call'
  activerecord (4.2.4) lib/active_record/connection_adapters/abstract/connection_pool.rb:653:in `call'
  activerecord (4.2.4) lib/active_record/migration.rb:377:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/callbacks.rb:29:in `block in call'
  activesupport (4.2.4) lib/active_support/callbacks.rb:88:in `__run_callbacks__'
  activesupport (4.2.4) lib/active_support/callbacks.rb:778:in `_run_call_callbacks'
  activesupport (4.2.4) lib/active_support/callbacks.rb:81:in `run_callbacks'
  actionpack (4.2.4) lib/action_dispatch/middleware/callbacks.rb:27:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/reloader.rb:73:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/remote_ip.rb:78:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
  web-console (2.2.1) lib/web_console/middleware.rb:31:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
  railties (4.2.4) lib/rails/rack/logger.rb:38:in `call_app'
  railties (4.2.4) lib/rails/rack/logger.rb:20:in `block in call'
  activesupport (4.2.4) lib/active_support/tagged_logging.rb:68:in `block in tagged'
  activesupport (4.2.4) lib/active_support/tagged_logging.rb:26:in `tagged'
  activesupport (4.2.4) lib/active_support/tagged_logging.rb:68:in `tagged'
  railties (4.2.4) lib/rails/rack/logger.rb:20:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/request_id.rb:21:in `call'
  rack (1.6.4) lib/rack/methodoverride.rb:22:in `call'
  rack (1.6.4) lib/rack/runtime.rb:18:in `call'
  activesupport (4.2.4) lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
  rack (1.6.4) lib/rack/lock.rb:17:in `call'
  actionpack (4.2.4) lib/action_dispatch/middleware/static.rb:116:in `call'
  rack (1.6.4) lib/rack/sendfile.rb:113:in `call'
  railties (4.2.4) lib/rails/engine.rb:518:in `call'
  railties (4.2.4) lib/rails/application.rb:165:in `call'
  rack (1.6.4) lib/rack/content_length.rb:15:in `call'
  puma (2.11.1) lib/puma/server.rb:507:in `handle_request'
  puma (2.11.1) lib/puma/server.rb:375:in `process_client'
  puma (2.11.1) lib/puma/server.rb:262:in `block in run'
  puma (2.11.1) lib/puma/thread_pool.rb:104:in `call'
  puma (2.11.1) lib/puma/thread_pool.rb:104:in `block in spawn_thread'

fcd1 commented:
More info about the above error:

#should_render?(xml_translation_logic, df_data) is called with xml_translation_logic set to:
["element", "mods:titleInfo"]
and app/models/concerns/digital_object/xml_datastream_rendering.rb:22 attempts to use the following:
xml_translation_logic['render_if'].present?
which, I assume, generates the:
TypeError (no implicit conversion of String into Integer)

The Item generating the error only has one field populated, the Title Sort Portion field, so it's the simplest test case and should be easy to reproduce.
